nipy/algorithms/clustering/gmm.py

```diff
@@ -217,8 +218,8 @@ def plot2D(x, my_gmm, z=None, with_dots=True, log_scale=False, mpaxes=None,
         if z is None:
             plt.plot(x[:, 0], x[:, 1], 'o')
         else:
-            hsv = plt.cm.hsv(range(256))
-            col = hsv[range(0, 256, 256 // int(z.max() + 1))]
+            hsv = plt.cm.hsv(list(range(256)))
+            col = hsv[list(range(0, 256, 256 // int(z.max() + 1)))]
```

Comment:
Okay on `hsv`. What about indexing numpy arrays? There are some other ones where this also happens.

An alternative is to use slicing instead of an iterator:

    col = hsv[slice(0, 256, 256 // int(z.max() + 1))]

or

    col = hsv[::(256 // int(z.max() + 1))]